src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowDiversityV2Spec.groovy

```diff
@@ -73,8 +76,27 @@ class FlowDiversityV2Spec extends HealthCheckSpecification {
         }
         allInvolvedIsls.unique(false) == allInvolvedIsls
 
-        cleanup: "Delete flows"
+        and: "Flows' histories contain 'groupId' information"
+        [flow2, flow3].each {//flow1 had no diversity at the time of creation
+            assert northbound.getFlowHistory(it.flowId).find { it.action == CREATE_ACTION }.dumps
+                    .find { it.type == "stateAfter" }?.groupId
+        }
+
+        when: "Delete flows"
         [flow1, flow2, flow3].each { flowHelperV2.deleteFlow(it.flowId) }
+        def flowsDeleted = true
+
+        then: "Flows' histories contain 'groupId' information in 'delete' operation"
+        [flow1, flow2, flow3].each {
+            verifyAll(northbound.getFlowHistory(it.flowId).find { it.action == DELETE_ACTION }.dumps) {
+                it.find { it.type == "stateBefore" }?.groupId
+                !it.find { it.type == "stateAfter" }?.groupId
+            }
+        }
+
+        cleanup:
+        !flowsDeleted && [flow1, flow2, flow3].each { flowHelperV2.deleteFlow(it.flowId) }
```

Comment:
it can fail in future
for example: when the system was able to delete the first two flow and wasn't able to delete the third flow

To tell the truth I have no idea how to handle it and at the same do not add extra complexity

Reply:
You can apply the question 'what if system wasn't able to delete a flow' to literally any test in our framework. Not sure what you expect to see. We assume that test deletion is working properly and also test it in separate flowcrud tests

Reply:
I meant that a flow wasn't deleted because it was in the in_progress state(rare case).
I agree, we should assume that the 'delete' operation works fine

Reply:
This mostly appears in tests where we cause flows to reroute without precisely tracking their reroute status (brought down controller or changed isls state etc.). In this particular test we only do flow create via flowHelper, which ensures no 'in-progress' state.So I think this test is the least possible test to have in-progress problems out of most of our tests.